src/Catty/Functions/Text/RegularExpressionFunction.swift

```diff
@@ -0,0 +1,79 @@
+/**
+ *  Copyright (C) 2010-2020 The Catrobat Team
+ *  (http://developer.catrobat.org/credits)
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Affero General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or (at your option) any later version.
+ *
+ *  An additional term exception under section 7 of the GNU Affero
+ *  General Public License, version 3, is available at
+ *  (http://developer.catrobat.org/license_additional_term)
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see http://www.gnu.org/licenses/.
+ */
+
+class RegularExpressionFunction: DoubleParameterStringFunction {
+    static var tag = "REGEX"
+    static var name = "regex"
+    static var defaultValue = ""
+    static var requiredResource = ResourceType.noResources
+    static var isIdempotent = true
+    static let position = 85
+
+    func tag() -> String {
+        return type(of: self).tag
+    }
+
+    func firstParameter() -> FunctionParameter {
+        return .string(defaultValue: " an? ([^ .]+)")
+    }
+
+    func secondParameter() -> FunctionParameter {
+        return .string(defaultValue: "I am a panda")
+    }
+
+    func value(firstParameter: AnyObject?, secondParameter: AnyObject?) -> String {
+        let pattern = type(of: self).interpretParameter(parameter: firstParameter)
+        let longText = type(of: self).interpretParameter(parameter: secondParameter)
+
+        return self.regularExpression(pattern: pattern, longText: longText)
+    }
+
+    func formulaEditorSections() -> [FormulaEditorSection] {
+        return [.math(position: type(of: self).position)]
+    }
+
+    func regularExpression(pattern: String, longText: String) -> String {
+        var finalResult = RegularExpressionFunction.defaultValue
+        let regexOptions: NSRegularExpression.Options = []
```

Comment:
Please verify whether your implementation matches the behavior of Android implementation in regards of multi-line "long texts". Please have a look at https://github.com/Catrobat/Catroid/blob/f9763f03dcf15049a69d9173417c927e1c305093/catroid/src/main/java/org/catrobat/catroid/formulaeditor/FormulaElement.java#L635 and see the options `DOTALL` and `MULTILINE`:

- https://docs.oracle.com/javase/7/docs/api/java/util/regex/Pattern.html#DOTALL
- https://docs.oracle.com/javase/7/docs/api/java/util/regex/Pattern.html#MULTILINE

Please add another "multi-line test" to your test class and check whether it is the same behavior. Consider adding those options to your implementation as well (after the tests fails).